web/src/AlertPane.tsx

```diff
@@ -33,7 +33,12 @@ class AlertResource {
   }
 
   public hasAlert() {
-    return this.podStatusIsError() || this.podRestarted() || this.buildFailed()
+    return (
+      this.podStatusIsError() ||
```

Comment:
i'm worried that the hasAlert() codepath and the render alerts codepath are so different, and likely to get out of sync.

have you considered having a getAlertElements() that returns what's currently in AlertPane's errorElements? then hasAlerts() just returns `getAlertElements().length > 0`, and we're back to a single source of truth

Reply:
That's a good idea, I'll add it as a TODO and get to it today.